### PR TITLE
Nim pretty and add void

### PR DESCRIPTION
**Pull request by kdb424**
Added void to distro list, added other arch based distros to `LacksDevPackages`, and used `nim pretty` on distros.nim.

### CODE REVIEW

File: lib/pure/distros.nim [hunk 1]
```diff
@@ -173,7 +177,8 @@ proc detectOsImpl(d: Distribution): bool =
       case d
       of Distribution.Gentoo:
         result = ("-" & $d & " ") in uname()
-      of Distribution.Elementary, Distribution.Ubuntu, Distribution.Debian, Distribution.Fedora,
+      of Distribution.Elementary, Distribution.Ubuntu, Distribution.Debian,
+          Distribution.Fedora,
         Distribution.OpenMandriva, Distribution.CentOS, Distribution.Alpine,
         Distribution.Mageia, Distribution.Zorin:
```
Comment (konsumlamm) [inline lib/pure/distros.nim]: That's some inconsistent formatting...

File: lib/pure/distros.nim [hunk 2]
```diff
@@ -187,14 +192,17 @@ proc detectOsImpl(d: Distribution): bool =
         # Check if this is a Nix build or NixOS environment
         result = existsEnv("NIX_BUILD_TOP") or existsEnv("__NIXOS_SET_ENVIRONMENT_DONE")
       of Distribution.OpenSUSE:
-        result = "suse" in toLowerAscii(uname()) or "suse" in toLowerAscii(release())
+        result = "suse" in toLowerAscii(uname()) or "suse" in toLowerAscii(
+            release())
```
Comment (konsumlamm) [inline lib/pure/distros.nim]: It's not really worth it to start a new line here.

File: lib/pure/distros.nim (hunk 1 above)
Comment (kdb424) [reply lib/pure/distros.nim]: This is what `nimpretty lib/pure/distros.nim` produced. I'll format it back, but seems like a bug in that.

File: lib/pure/distros.nim (hunk 2 above)
Comment (kdb424) [reply lib/pure/distros.nim]: Same as above. I'll reformat it back.

File: lib/pure/distros.nim [hunk 3]
```diff
@@ -173,9 +177,9 @@ proc detectOsImpl(d: Distribution): bool =
       case d
       of Distribution.Gentoo:
         result = ("-" & $d & " ") in uname()
-      of Distribution.Elementary, Distribution.Ubuntu, Distribution.Debian, Distribution.Fedora,
-        Distribution.OpenMandriva, Distribution.CentOS, Distribution.Alpine,
-        Distribution.Mageia, Distribution.Zorin:
+      of Distribution.Elementary, Distribution.Ubuntu, Distribution.Debian,
+        Distribution.Fedora, Distribution.OpenMandriva, Distribution.CentOS,
+	Distribution.Alpine, Distribution.Mageia, Distribution.Zorin:
```
Comment (konsumlamm) [inline lib/pure/distros.nim]:
```suggestion
      of Distribution.Elementary, Distribution.Ubuntu, Distribution.Debian,
        Distribution.Fedora, Distribution.OpenMandriva, Distribution.CentOS,
	  Distribution.Alpine, Distribution.Mageia, Distribution.Zorin:
```

**Comment by kdb424** [comment]
I'll be more careful in the future now that I know nimpretty has some bugs obviously. Thank you for helping me sort those out.

File: lib/pure/distros.nim [hunk 4]
```diff
@@ -195,6 +199,8 @@ proc detectOsImpl(d: Distribution): bool =
         result = ("sun" in uname) or ("solaris" in uname)
       of Distribution.Haiku:
         result = defined(haiku)
+      of Distribution.Void:
+        result = "void" in osReleaseID()
```
Comment (dom96) [inline lib/pure/distros.nim]: Why not just add this to the branch above along with Fedora, Elementary, etc? It will work just as well

**Review by dom96 — APPROVED** [review]
Looks good, but see comment :)

File: lib/pure/distros.nim (hunk 4 above)
Comment (kdb424) [reply lib/pure/distros.nim]: Curious to why "arch" is split out. I wasn't sure of the specifics, so I added it that way. If there is no great reason, I can just squash them all in (rhel, arch, artix, void) to be up with the rest.

Comment (dom96) [reply lib/pure/distros.nim]: because it would check for "archlinux" which isn't the same as "arch", no?

Comment (kdb424) [reply lib/pure/distros.nim]: Good point. Resolving your requested change. Thank you for your feedback to something that was obvious.

**Pull request merged**